bin/setupReplication.js

```diff
@@ -0,0 +1,431 @@
+            this._targetBucket;
+        const params = {
+            Bucket: bucket,
+            VersioningConfiguration: {
+                Status: 'Enabled',
+            },
+        };
+        this._s3Clients[where].putBucketVersioning(params, (err, res) => {
+            if (err) {
+                this._log.error('error enabling versioning', {
+                    error: err,
+                    method: '_SetupReplication._enableVersioning',
+                    errorMessage: err.message,
+                    errorStack: err.stack,
+                });
+                return cb(err);
+            }
+            this._log.debug('Versioning enabled', {
+                bucket: where,
+                response: res,
+                method: '_enableVersioning',
+            });
+            return cb(null, res);
+        });
+    }
+
+    _attachResourcePolicy(policyArn, roleName, where, cb) {
+        const params = {
+            PolicyArn: policyArn,
+            RoleName: roleName,
+        };
+        this._iamClients[where].attachRolePolicy(params, (err, res) => {
+            if (err) {
+                this._log.error('error attaching resource policy', {
+                    error: err,
+                    method: '_SetupReplication._attachResourcePolicy',
+                    errorMessage: err.message,
+                    errorStack: err.stack,
+                });
+                return cb(err);
+            }
+            this._log.debug('Attached resource policy', {
+                bucket: where,
+                response: res,
+                method: '_attachResourcePolicy',
+            });
+            return cb(null, res);
+        });
+    }
+
+    _enableReplication(roleArns, cb) {
+        const params = {
+            Bucket: this._sourceBucket,
+            ReplicationConfiguration: {
+                Role: roleArns,
+                Rules: [{
+                    Destination: {
+                        Bucket: `arn:aws:s3:::${this._targetBucket}`,
+                    },
+                    Prefix: '',
+                    Status: 'Enabled',
+                }],
+            },
+        };
+        this._s3Clients.source.putBucketReplication(params, (err, res) => {
+            if (err) {
+                this._log.error('error enabling replication', {
+                    error: err,
+                    method: '_SetupReplication._enableReplication',
+                    errorMessage: err.message,
+                    errorStack: err.stack,
+                });
+                return cb(err);
+            }
+            this._log.debug('Bucket replication enabled', {
+                response: res,
+                method: '_enableReplication',
+            });
+            return cb(null, res);
+        });
+    }
+
+    _parallelTasks(cb) {
+        return async.parallel({
+            sourceBucket: next => this._createBucket('source', next),
```

Comment:
I think using an object here would be better because when calling `async.series`, I'm using the key names for accessing corresponding data. I think it's more descriptive this way vs accessing these from an array